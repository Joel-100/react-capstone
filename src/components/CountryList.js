import React from 'react'

const CountryList = () => {
  return (
    <div>CountryList</div>
  )
}

export default CountryList