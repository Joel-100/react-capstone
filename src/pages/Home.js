import React from 'react';
import africaImg from '../assets/africa.png';

const Home = () => (
  <div>
    <div className="flex p-5 items-center border-b">
      <img src={africaImg} alt="Africa map" className="w-[60%] opacity-60" />
      <div className="ml-6">
        <h3 className="text-3xl font-mono text-white font-bold">Africa</h3>
        <p className="text-xl text-white">1,393.7 billion</p>
      </div>
    </div>
    <div className="text-white my-4">
      <h3 className="ml-8">Stats by Country</h3>
    </div>
  </div>
);

export default Home;
