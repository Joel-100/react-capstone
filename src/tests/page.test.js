import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store/store';
import Home from '../pages/Home';
import CountryList from '../components/CountryList';

describe('Page tests', () => {
  test('Render Home Page', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Render CountryList Component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <CountryList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
