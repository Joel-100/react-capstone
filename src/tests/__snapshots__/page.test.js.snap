// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page tests Render CountryList Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page tests Render Home Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex p-5 items-center border-b"
        >
          <img
            alt="Africa map"
            class="w-[60%] opacity-60"
            src="africa.png"
          />
          <div
            class="ml-6"
          >
            <h3
              class="text-3xl font-mono text-white font-bold"
            >
              Africa
            </h3>
            <p
              class="text-xl text-white"
            >
              1,393.7 billion
            </p>
          </div>
        </div>
        <div
          class="text-white my-4"
        >
          <h3
            class="ml-8"
          >
            Stats by Country
          </h3>
        </div>
        <div
          class="flex flex-wrap"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex p-5 items-center border-b"
      >
        <img
          alt="Africa map"
          class="w-[60%] opacity-60"
          src="africa.png"
        />
        <div
          class="ml-6"
        >
          <h3
            class="text-3xl font-mono text-white font-bold"
          >
            Africa
          </h3>
          <p
            class="text-xl text-white"
          >
            1,393.7 billion
          </p>
        </div>
      </div>
      <div
        class="text-white my-4"
      >
        <h3
          class="ml-8"
        >
          Stats by Country
        </h3>
      </div>
      <div
        class="flex flex-wrap"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
